### ISSUE TITLE
Event text is cut off in day/week view

### ISSUE THREAD

**Issue by cole8888** [nextcloud/calendar#2644]
### Steps to reproduce
1. Update to Nextcloud 20.0.0 beta
2. Create an event with a time duration which will cause the text to get cut off. (0mins, 20mins, 40mins, 50mins, etc...)

### Expected behaviour
I expected that despite the time duration of the event, the time and title should be visible. This was the case when using Nextcloud 19.

### Actual behaviour
Various event time duration ranges cause the event text and or time to be completely hidden or cut off.
![image](https://user-images.githubusercontent.com/32819560/95914685-7978d580-0d7c-11eb-910f-ec5fe4139a68.png)

### Calendar app
**App version:** 2.1.2

**CalDAV-clients used:** N/A

### Client configuration
**Browser:** Firefox 81.0

**Operating system:** Kubuntu 20.04

### Server configuration

**Operating system**: Ubuntu Server 20.04
**Web server:** Nginx

**Database:** MariaDB

**PHP version:** 7.4.3

**Nextcloud Version:** 20.0.0 (beta)

**Updated from an older installed version or fresh install:** Upgraded from 19

**List of activated apps:**
```
Enabled:
  - activity: 2.13.1
  - admin_audit: 1.10.0
  - analytics: 3.0.0
  - breezedark: 19.0.5
  - calendar: 2.1.2
  - cloud_federation_api: 1.3.0
  - contactsinteraction: 1.1.0
  - dav: 1.16.0
  - drawio: 0.9.7
  - extract: 1.2.4
  - federatedfilesharing: 1.10.1
  - files: 1.15.0
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.0
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - groupfolders: 8.1.0
  - keeweb: 0.6.3
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - metadata: 0.12.0
  - oauth2: 1.8.0
  - photos: 1.2.0
  - previewgenerator: 3.0.1
  - provisioning_api: 1.10.0
  - richdocuments: 3.7.4
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - twofactor_totp: 5.0.0
  - user_status: 1.0.0
  - video_converter: 0.1.4
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
```

**Nextcloud configuration:**

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "MY_DOMAIN_DOT_COM",
            "192.168.2.100"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "20.0.0.9",
        "overwrite.cli.url": "https:\/\/MY_DOMAIN_DOT_COM",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "mysql.utf8mb4": true,
        "installed": true,
        "auth.bruteforce.protection.enabled": true,
        "trashbin_retention_obligation": "auto",
        "versions_retention_obligation": "disabled",
        "updater.release.channel": "beta",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "enable_previews": true,
        "maintenance": false,
        "filelocking.enabled": true,
        "integrity.check.disabled": true,
        "theme": "",
        "loglevel": 2,
        "app_install_overwrite": [
            "keeweb",
            "files_clipboard",
            "occweb",
            "group_everyone",
            "issuetemplate",
            "files_reader",
            "groupfolders",
            "ocsms",
            "video_converter",
            "files_photospheres",
            "audioplayer",
            "extract",
            "metadata",
            "richdocuments",
            "discoursesso",
            "onlyoffice",
            "drawio",
            "breezedark",
            "documentserver_community"
        ],
        "updater.secret": "***REMOVED SENSITIVE VALUE***",
        "has_rebuilt_cache": true,
        "onlyoffice": {
            "verify_peer_off": true,
            "jwt_header": "AuthorizationJwt"
        }
    }
}
```

#### Web server error log (e.g. /var/log/apache)
```
Too large to paste
```

#### Log file (data/nextcloud.log)
```
Too large to paste
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
Content Security Policy: Couldn’t process unknown directive “default-src:”
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). content-fontface.js:50:25
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). content-script.js:40:65
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). content-script.js:40:65
No OC found main.js:1:323308
JQMIGRATE: Migrate is installed, version 1.4.1 main.js:1:917387
jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. main.js:1:1407020
$ is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. main.js:1:1407020
jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 2 main.js:1:1407020
[INFO] calendar: 
Initial settings:
	- AppVersion: 2.1.2
	- EventLimit: false
	- FirstRun: false
	- ShowWeekNumbers: false
	- ShowTasks: true
	- ShowWeekends: true
	- SkipPopover: true
	- SlotDuration: 00:30:00
	- TalkEnabled: false
	- TasksEnabled: false
	- Timezone: automatic
 
Object { app: "calendar", uid: "Cole" }
calendar.js:2:2164605
[DEBUG] calendar: The calendar app is using version 2.2019c of the timezone database 
Object { app: "calendar", uid: "Cole" }
calendar.js:2:2164534
Fallback to locale en calendar.js:2:1099917
[INFO] calendar: Updated moment locale: en 
Object { app: "calendar", uid: "Cole" }
calendar.js:2:2164605
Source map error: Error: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Resource URL: https://MY_DOMAIN_DOT_COM/nextcloud/core/js/dist/main.js?v=02e0a362-32
Source Map URL: main.js.map
jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 4 main.js:1:1407020
session heartbeat polling started main.js:1:711925
$ is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 2 main.js:1:1407020
jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 4 main.js:1:1407020
Source map error: Error: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Resource URL: https://MY_DOMAIN_DOT_COM/nextcloud/apps/calendar/js/calendar.js?v=02e0a362-32
Source Map URL: calendar.js.map
[DEBUG] unified-search: Unified Search initialized with the following providers 
Object { 0: {…}, 1: {…}, 2: {…}, 3: {…}, 4: {…}, 5: {…}, 6: {…}, app: "unified-search", uid: "Cole" }
unified-search.js:1:271869
jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 7 main.js:1:1407020
jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. main.js:1:1407020
jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 144 main.js:1:1407020
[DEBUG] calendar: Current user principal is /nextcloud/remote.php/dav/principals/users/Cole/ 
Object { app: "calendar", uid: "Cole" }
calendar.js:2:2164534
Source map error: Error: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Resource URL: https://MY_DOMAIN_DOT_COM/nextcloud/core/js/dist/unified-search.js?v=02e0a362-32
Source Map URL: unified-search.js.map
jQuery is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 4 main.js:1:1407020

```

**Comment by JonathanTreffler** [comment]
I have the same Problem:
I use Nextcloud Calendar 2.1.2 with Nextcloud 20.0.2.

I am not sure if the cause is the same, but the problem looks the same.

**Comment by alexanderdd** [comment]
I can confirm this bug is still present in calendar 2.1.3 with Nextcloud 20.0.4

@georgehrke can you change the title to "Event text is cut off in day/week view" and add the label "regression"? Maybe also assign a milestone?

Usability is significiantly decreased because of this. Nextcloud claims it puts a big emphasis on usability. This was reported three months ago.

**Comment by JonathanTreffler** [comment]
Maybe @jancborchardt could get involved.

The problem still exists in the current version and pretty much every event in my calendar is 45 Minutes long -> pretty much every event name is not visible.

![image](https://user-images.githubusercontent.com/28999431/105006657-3c854480-5a37-11eb-8668-53f516df9dfc.png)

**Comment by JonathanTreffler** [comment]
> can you change the title to "Event text is cut off in day/week view"

done

**Comment by alexanderdd** [comment]
@jancborchardt this is a serious UX issue and has been open for four months now.

**Comment by alexanderdd** [comment]
Related https://github.com/nextcloud/calendar/issues/2696

**Comment by JonathanTreffler** [comment]
> Who can do the triage? (so "to triage" is removed and "to develop" gets added?)

done

**Comment by JonathanTreffler** [comment]
(this is not really my corner of nextcloud developement, so i hope nobody gets mad that i took the liberty of changing the labels 🤣)

**Comment by mafgh** [comment]
Since #2696 was closed, adding
```
.fc-v-event {
    min-height: 2em;
}
```
to `nextcloud/apps/calendar/css/fullcalendar.scs` is a workaround for me.

**Comment by daCaPo** [comment]
> Since #2696 was closed, adding
> 
> ```
> .fc-v-event {
>     min-height: 2em;
> }
> ```
> 
> to `nextcloud/apps/calendar/css/fullcalendar.scs` is a workaround for me.

Still works with 2.2.1

**Comment by alexanderdd** [comment]
> > Since #2696 was closed, adding
> > ```
> > .fc-v-event {
> >     min-height: 2em;
> > }
> > ```
> > 
> > 
> > to `nextcloud/apps/calendar/css/fullcalendar.scs` is a workaround for me.
> 
> Still works with 2.2.1

That seems to be the solution for very short events. **Who can turn this into a pull request?**

Does anybody know **what to do about 45min events**? There is enough space to display event text, but the text is cut off, see screenshot by @JonathanTreffler above.

**Comment by mrtumnus** [comment]
> Does anybody know what to do about 45min events? There is enough space to display event text, but the text is cut off, see screenshot by @JonathanTreffler above.

The height at which events are condensed into a one-line format is defined in fullcalendar's [TimeCol.tsx](https://github.com/fullcalendar/fullcalendar/blob/v5.6.0/packages/timegrid/src/TimeCol.tsx#L33):

    config.timeGridEventCondensedHeight = 30;

I played around a bit, and it looks like a single line is around 24 pixels, so I chose 50 as an alternative.  This allowed anything less than two lines to be condensed into a single line.  Still not great, as we still have the full "0:00 XX - 0:00 XX" time display preceding the event text.  But better than not showing anything.

So, in summary, I worked around this by setting the following value in `nextcloud/apps/calendar/js/calendar.js`:

    am.timeGridEventCondensedHeight = 50;

It looks like this property went away in fullcalendar v5.7.0, so maybe there are some improvements in this regard incoming soon.

**Comment by mrtumnus** [comment]
Here's another workaround which I prefer a little better, along the same lines as what @mafgh proposed:

```
.fc-v-event {
        min-height: 4em;
}

.fc-v-event.fc-timegrid-event-condensed {
        min-height: 2em;
}
```

**Comment by JonathanTreffler** [comment]
I tested both css fixes above through the custom css app (not through editing the source code) and both did not work for me. Maybe its a screen size issue, but i will look into it further.

**Comment by mrtumnus** [comment]
> I tested both css fixes above through the custom css app (not through editing the source code) and both did not work for me. Maybe its a screen size issue, but i will look into it further.

What is the 'custom css app'?  I edited the source code to produce the screenshots above.

**Comment by JonathanTreffler** [comment]
> What is the 'custom css app'?

https://apps.nextcloud.com/apps/theming_customcss

**Comment by alexanderdd** [comment]
Can confirm that the method described by @mrtumnus works :smile: 
But in this specific case (note the times), the title can still not be read properly:
![Bildschirmfoto vom 2021-06-04 01-30-40](https://user-images.githubusercontent.com/28591861/120724864-f0faa100-c4d4-11eb-83e6-9d8649631860.png)
Nextcloud 20.0.10, Calendar 2.2.2 on Firefox 89.0 (let me know if you need more details).

**Comment by mrtumnus** [comment]
> But in this specific case (note the times), the title can still not be read properly:
Nice catch.  I'm not sure how to make continguous events appear as truly overlapping events would.  Maybe this approach is too naïve.

**Comment by mrtumnus** [comment]
My pull request in #3121 was approved & merged, so maybe the case @alexanderdd raised can be addressed separately.

**Comment by cole8888** [comment]
Original issue appears to be fixed in version 2.3.0, thanks :)

![image](https://user-images.githubusercontent.com/32819560/123898691-25c71000-d93c-11eb-9d5f-9b8242e1a297.png)

**Comment by mrtumnus** [comment]
Awesome.  Looks like they went with a minimum height equivalent of 30 minutes.

**Comment by alexanderdd** [comment]
Yes, original issue here is fixed in calendar 2.3.0, thanks :partying_face: Thank you for your work @mrtumnus !!!!

This  https://github.com/nextcloud/calendar/issues/2644#issuecomment-854252929 is still a problem so I opened a new issue https://github.com/nextcloud/calendar/issues/3274

And I disagree with a minimum height equivalent of 30 minutes so I opened this https://github.com/nextcloud/calendar/issues/3275
